### DISCUSSION TITLE
Persistent vault filter in browser extension

### DISCUSSION THREAD

**Discussion by infogulch** [bitwarden/clients#12473] [Password Manager]
### Code Contribution Proposal

Hi!

I am interested in using a [Default/Persistent Vault Selection](https://community.bitwarden.com/t/option-to-set-a-default-persistent-vault-filter/43641/23) feature, enough that I may be willing to contribute the implementation. I found no proposals like it ~~but this one is mine~~ so here is mine.

In my opinion this should be a per-client-instance setting, not synced at all. This gives users the maximum flexibility at the low cost of one-time configuration per client. Another benefit is that it makes the implementation much easier. :) (The benefit to complexity tradeoff for this is so weighted towards do-not-sync that I don't think it would be worth implementing even given unlimited implementation time.)

The best version of this feature would allow users to select any subset of vaults as default by making the All Vaults dropdown a multi-select. (Technically I think I would store a list of unselected vaults, so newly added vaults would be visible on all existing clients by default. I think this would be less confusing to users that forgot they are using the feature.) I'm keen to attempt this, but maybe that's already too much for a first contribution since the clients do not currently support searching across a subset of vaults (just one or all) as far as I'm aware. If not, perhaps this could be included in a followup proposal.

I suspect it would be unrealistic to contribute this feature to all clients at once, so I plan to target the browser extension client first.

Would the Bitwarden team consider accepting such a contribution? Of course I am open to feature/scope changes with respect to the above proposal.

Thanks!
Joe

**Comment by HenryJrCondolo** [comment]
This function would be perfect; there are already collections that separate passwords a bit, but without any way to have a fixed/persistent filter, it's like just having another folder in reality. Other password managers, such as 1Password, have these options available in their settings, allowing you to have a personal vault and a family or work vault. Depending on what you need on the device, you can select one of the vaults or both. This method is more effective and eliminates the need to apply filters, enabling you to search for only work-related passwords without seeing personal passwords.

Hopefully, a similar function can be implemented.

**Comment by dwbit** [comment]
Closing this one out as the team has been working on vault filter persistence, but feel free to add any enhancements to the [feature request category](https://community.bitwarden.com/c/feature-requests/pm-feature-requests/55) of the Bitwarden Forums.

**Reply by infogulch** [reply]
Glad to hear it, thanks for updating this discussion!

**Reply by infogulch** [reply]
Is someone on the team still working on this?

**Reply by infogulch** [reply]
Wait has this been done since febuary? https://github.com/bitwarden/clients/pull/13303
